From bd5e4c4611b4dabad73b78284cd6502157852480 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Mon, 2 Mar 2020 15:37:40 -0800
Subject: [PATCH] dlopen system tbb

---
 RcppParallel/R/build.R | 33 +++++++++++++++++----------------
 1 file changed, 17 insertions(+), 16 deletions(-)

diff --git a/RcppParallel/R/build.R b/RcppParallel/R/build.R
index 95cbf5e..7c0c8fd 100644
--- a/RcppParallel/R/build.R
+++ b/RcppParallel/R/build.R
@@ -64,22 +64,23 @@ tbbLdFlags <- function() {
 
 # Determine the platform-specific path to the TBB library
 tbbLibPath <- function(suffix = "") {
-   sysname <- Sys.info()['sysname']
-   tbbSupported <- list(
-      "Darwin" = paste("libtbb", suffix, ".dylib", sep = ""), 
-      "Linux" = paste("libtbb", suffix, ".so.2", sep = ""), 
-      "Windows" = paste("tbb", suffix, ".dll", sep = ""),
-      "SunOS" = paste("libtbb", suffix, ".so", sep = "")
-   )
-   if ((sysname %in% names(tbbSupported)) && !isSparc()) {
-      libDir <- "lib/"
-      if (sysname == "Windows")
-         libDir <- paste(libDir, .Platform$r_arch, "/", sep="")
-      system.file(paste(libDir, tbbSupported[[sysname]], sep = ""), 
-                  package = "RcppParallel")
-   } else {
-      NULL
-   }
+   ## sysname <- Sys.info()['sysname']
+   ## tbbSupported <- list(
+   ##    "Darwin" = paste("libtbb", suffix, ".dylib", sep = ""), 
+   ##    "Linux" = paste("libtbb", suffix, ".so.2", sep = ""), 
+   ##    "Windows" = paste("tbb", suffix, ".dll", sep = ""),
+   ##    "SunOS" = paste("libtbb", suffix, ".so", sep = "")
+   ## )
+   ## if ((sysname %in% names(tbbSupported)) && !isSparc()) {
+   ##    libDir <- "lib/"
+   ##    if (sysname == "Windows")
+   ##       libDir <- paste(libDir, .Platform$r_arch, "/", sep="")
+   ##    system.file(paste(libDir, tbbSupported[[sysname]], sep = ""), 
+   ##                package = "RcppParallel")
+   ## } else {
+   ##    NULL
+   ## }
+   paste("/usr/lib64/libtbb", suffix, ".so.2", sep = "")
 }
 
 isSparc <- function() {
-- 
2.25.1

